dbfit-java/core/src/test/java/dbfit/environment/TypeSpecifierTest.java

```diff
@@ -0,0 +1,35 @@
+package dbfit.environment;
+
+import org.junit.*;
+import static org.junit.Assert.*;
+
+import java.sql.SQLException;
+import java.util.Map;
+
+import dbfit.api.DBEnvironment;
+
+import dbfit.util.TypeSpecifier;
+
+public class TypeSpecifierTest {
+
+    private class DummyTypeSpecifier implements TypeSpecifier {
+        public Object specify(Object o) throws SQLException {
+            return new String("abcdef");
+        }
+    }
+
+    @Test
+    public void castTypeTest() {
+        DBEnvironment env = dbfit.api.DbEnvironmentFactory.newEnvironmentInstance("Dummy");
+        env.setTypeSpecifier(java.util.Date.class, new DummyTypeSpecifier());
+        Map<Class<?>, TypeSpecifier> tsm = env.getTypeSpecifierMap();
+        TypeSpecifier ts = tsm.get(java.util.Date.class);
+        String s = null;
+        try {
+            s = (String) ts.specify(new java.util.Date());
```

Comment:
>> Maybe we should rather start from unit testing around StatementExecution as it seems closer to the problem we want to solve.

> How about using a Derby or HSQLDB DB for this (unless there's some kind of mocking available)?

We have Mockito already available (it's in use in some of the tests). But the thing is currently we're calling many things in this test and I'm not exactly sure what exactly we're testing. (It's good to have smaller and more specific tests - might be several ones /if it fails - it's easier to isolate the reason/).